vpr/src/place/weighted_median_move_generator.cpp

```diff
@@ -0,0 +1,178 @@
+#include "weighted_median_move_generator.h"
+#include "globals.h"
+#include <algorithm>
+#include "math.h"
+
+static void get_bb_cost_for_net_excluding_block(ClusterNetId net_id, t_bb_cost* coords, ClusterBlockId block_id, ClusterPinId moving_pin_id, const PlacerCriticalities* criticalities, bool& skip_net);
+
+e_create_move WeightedMedianMoveGenerator::propose_move(t_pl_blocks_to_be_moved& blocks_affected, e_move_type& /*move_type*/, float rlim, const t_placer_opts& placer_opts, const PlacerCriticalities* criticalities) {
+    auto& place_ctx = g_vpr_ctx.placement();
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+
+    auto& place_move_ctx = g_placer_ctx.mutable_move();
+
+    /* Pick a random block to be swapped with another random block.   */
+    ClusterBlockId b_from = pick_from_block();
+
+    if (!b_from) {
+        return e_create_move::ABORT; //No movable block found
+    }
+
+    t_pl_loc from = place_ctx.block_locs[b_from].loc;
+    auto cluster_from_type = cluster_ctx.clb_nlist.block_type(b_from);
+    auto grid_from_type = g_vpr_ctx.device().grid[from.x][from.y].type;
+    VTR_ASSERT(is_tile_compatible(grid_from_type, cluster_from_type));
+
+    /* Calculate the Edge weighted median region */
+    t_pl_loc to;
+
+    t_bb_cost coords;
+    t_bb limit_coords;
+
+    //clear the vectors that saves X & Y coords
+    //reused to save allocation time
+    place_move_ctx.X_coord.clear();
+    place_move_ctx.Y_coord.clear();
+
+    //true if the net is a feedback from the block to itself
```

Comment:
Does this mean the net has *only* terminals that connect to this one block, or does it mean the net has *some* terminals that connect the a block output back to one or more block inputs.  Precisely define skip_net.

Reply:
It means that this net's terminals are all connected to the same block. Edited the comment by this meaning to be more clear.